bitcoin/src/blockdata/witness.rs

```diff
@@ -361,14 +361,14 @@ mod test {
 
     #[test]
     fn consensus_serialize() {
-        let el_0 = hex!("03d2e15674941bad4a996372cb87e1856d3652606d98562fe39c5e9e7e413f2105");
-        let el_1 = hex!("000000");
+        let el_0 = hex!("03d2e15674941bad4a996372cb87e1856d3652606d98562fe39c5e9e7e413f2105").to_vec();
+        let el_1 = hex!("000000").to_vec();
 
         let mut want_witness = Witness::default();
         want_witness.push(&el_0);
         want_witness.push(&el_1);
 
-        let vec = vec![el_0.clone(), el_1.clone()];
+        let vec = vec![el_0, el_1];
```

Comment:
This is later used in a function `let got_witness = Witness::from_slice(&vec);`.  Can't this just be a slice the begin with?

Reply:
If we need to rebase this PR we could fix this. I'm a little surprised the clippy "unnecesary vec" lint didn't notice it.